PowerEditor/installer/themes/DarkModeDefault.xml

```diff
@@ -645,21 +645,21 @@ License:             GPL2
             <WordsStyle name="STRING" styleID="2" fgColor="DCDCCC" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
             <WordsStyle name="STRING EOL" styleID="3" fgColor="808080" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
-            <WordsStyle name="PROPERTY NAME" styleID="4" fgColor="8CD0D3" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
-            <WordsStyle name="ESCAPE SEQUENCE" styleID="5" fgColor="0000FF" bgColor="3F3F3F" fontName="" fontStyle="1" fontSize="" />
-            <WordsStyle name="LINE COMMENT" styleID="6" fgColor="008000" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
-            <WordsStyle name="BLOCK COMMENT" styleID="7" fgColor="008000" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
-            <WordsStyle name="OPERATOR" styleID="8" fgColor="E3CEAB" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
-            <WordsStyle name="URI" styleID="9" fgColor="FFFF80" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
-            <WordsStyle name="COMPACT IRI" styleID="10" fgColor="0000FF" bgColor="3F3F3F" fontName="" fontStyle="1" fontSize="" />
-            <WordsStyle name="KEYWORD" styleID="11" fgColor="18AF8A" bgColor="3F3F3F" fontName="" fontStyle="1" fontSize="" keywordClass="instre1"/>
-            <WordsStyle name="LD KEYWORD" styleID="12" fgColor="FF0000" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" keywordClass="instre2" />
-            <WordsStyle name="ERROR" styleID="13" fgColor="FFFF80" bgColor="FF0000" fontName="" fontStyle="0" fontSize="" />		
-		</LexerType>
+            <WordsStyle name="PROPERTY NAME" styleID="4" fgColor="CEDF99" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
+            <WordsStyle name="ESCAPE SEQUENCE" styleID="5" fgColor="DCA3A3" bgColor="3F3F3F" fontName="" fontStyle="1" fontSize="" />
+            <WordsStyle name="LINE COMMENT" styleID="6" fgColor="7F9F7F" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
+            <WordsStyle name="BLOCK COMMENT" styleID="7" fgColor="7F9F7F" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
+            <WordsStyle name="OPERATOR" styleID="8" fgColor="9F9D6D" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
+            <WordsStyle name="URI" styleID="9" fgColor="FFCFAF" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
+            <WordsStyle name="COMPACT IRI" styleID="10" fgColor="FFCFAF" bgColor="3F3F3F" fontName="" fontStyle="1" fontSize="" />
+            <WordsStyle name="KEYWORD" styleID="11" fgColor="DFC47D" bgColor="3F3F3F" fontName="" fontStyle="1" fontSize="" keywordClass="instre1"/>
+            <WordsStyle name="LD KEYWORD" styleID="12" fgColor="DFC47D" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" keywordClass="instre2" />
+            <WordsStyle name="ERROR" styleID="13" fgColor="EDD6ED" bgColor="3F3F3F" fontName="" fontStyle="0" fontSize="" />
+        </LexerType>
```

Comment:
The dark mode JSON highlighting looks good  to me:

![image](https://user-images.githubusercontent.com/90293/164125429-706f58a6-d8a8-4c12-8b41-680c20b6f810.png)

Could you show the screenshots before/after your modification please?

Reply:
Sure - the main issues are with lesser used features - i.e. [LD Keywords](https://w3c.github.io/json-ld-syntax/#syntax-tokens-and-keywords), [IRI's](https://w3c.github.io/json-ld-syntax/#base-iri) and comments.

Before:
![image](https://user-images.githubusercontent.com/5821575/164176876-e9feea02-3732-4421-8011-13ca5ba7f87a.png)

After:
![image](https://user-images.githubusercontent.com/5821575/164177122-1d06522f-6ba4-4746-bbca-169db8327054.png)

I noticed that the colours used in the json scheme didn't align with those used in all the other schemes (Java/C/JavaScript etc.) so I updated them to match for consistency.

Reply:
OK, I see. Thank you for your demo.
However, `// Foo`; `/* Foo */` & `BROKEN` should keep their old colour scheme for warning users the errors.

Reply:
I've reverted the error colour as requested.

![image](https://user-images.githubusercontent.com/5821575/164783447-a22dfce0-0151-433b-a2e6-071ee472867a.png)


This has identified that the `LINE COMMENT` and `BLOCK COMMENT` colours are actually ignored (and I'm guessing the `ESCAPE SEQUENCE` colour too).

Looking at the Lexer code, it would appear that you need to turn comment (and escape sequence) tokenisation on for them to be usable, as they are configurable options.

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/292c057f655ec5e691d92045ab78c24c01b8b102/lexilla/lexers/LexJSON.cxx#L120-L132

I'm not sure if those properties need to be set at compile time or runtime. If it's runtime, I'd like to work out how to turn them on. If it's compile time, I might open an issue over the weekend requesting that the comment option is turned on. If you're happy for them to be, I'll look to see if I can work out where in the N++ code the option needs configuring.

Reply:
The option is set on compile time, but I believe with help of PythonScript plugin, you can set it at runtime.
@alankilborn Could you give him a hand please?

Reply:
I've found the code where this would be set - `ScintillaEditView::setJsonLexer()`.

https://github.com/notepad-plus-plus/notepad-plus-plus/blob/292c057f655ec5e691d92045ab78c24c01b8b102/PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp#L685-L715

Would you be happy to accept a _separate_ pull request turning `lexer.json.escape.sequence` and `lexer.json.allow.comments` on?

Reply:
@ianjoneill 
JSON doesn't contain the comment:
https://stackoverflow.com/questions/244777/can-comments-be-used-in-json

That's why this option is not enabled in Notepad++ codes.
A few users can enable it on runtime, but for the most of users it should be disabled. 
Hope you understand.

Reply:
That's fair enough - it's supported in JSON5 (as per https://stackoverflow.com/a/64907232), but not in standard JSON.

Reply:
@ianjoneill 
What do you think about this solution?
https://github.com/notepad-plus-plus/notepad-plus-plus/issues/11676#issuecomment-1127111440

Reply:
Yes - that sounds like a good course of action